iteratee/src/main/scala/io/finch/iteratee/AsyncEncode.scala

```diff
@@ -0,0 +1,37 @@
+package io.finch.iteratee
+
+import java.nio.charset.Charset
+
+import com.twitter.io.Buf
+import com.twitter.util.Future
+import io.finch.Application
+import io.iteratee.Enumerator
+
+trait AsyncEncode[A] {
+
+  type ContentType <: String
+
+  def apply(enumerator: Enumerator[Future, A], cs: Charset): Enumerator[Future, Buf]
+}
+
+object AsyncEncode extends EncodeInstances {
```

Comment:
Thinking about this more. Maybe it's an overkill?

Looks like we can go from `Enumerator[Future, A]` to `Enumerator[Future, Buf]` just by using `Encode.Aux[CT, A]` (it takes charset). Maybe we don't really need this type class? I understand it makes things symmetric, but I think if we remove it and rename `AsyncDecode` to something like `Enumerate` it should all line up pretty nicely.